### PR TITLE
Fail to add this event 'perf probe select_task_rq_fair:15'

### PR DESCRIPTION
**Pull request by shirishaganta**
Changed the event a/c to that in script

Signed-off-by: shirisha Ganta <shiganta@in.ibm.com>

### CODE REVIEW

**Comment by shirishaganta** [comment]
Before Patch:
------------------
[root@ltczz256-lp3 perf]# avocado run perf_duplicate_probe.py 
/usr/local/lib/python3.6/site-packages/avocado_framework-80.0-py3.6.egg/avocado/plugins/run.py:285: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp. 
  FutureWarning)
JOB ID     : 58b0d4308d74b1a106f4712cef9ba0a7fca76e61
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2020-07-20T23.56-58b0d43/job.log
 (1/1) perf_duplicate_probe.py:PerfProbe.test_probe: ERROR: Command 'perf probe select_task_rq_fair:15' failed.\nstdout: b''\nstderr: b"Probe point 'select_task_rq_fair:15' not found.\n  Error: Failed to add events.\n"\nadditional_info: None (127.51 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2020-07-20T23.56-58b0d43/results.html
JOB TIME   : 127.64 s
[root@ltczz256-lp3 perf]# 

After Patch:
--------------
[root@ltczz256-lp3 perf]# avocado run perf_duplicate_probe.py
/usr/local/lib/python3.6/site-packages/avocado_framework-80.0-py3.6.egg/avocado/plugins/run.py:285: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp. 
  FutureWarning)
JOB ID     : 4f358eb1611fb80f165fcd26169732045713807d
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2020-07-21T01.29-4f358eb/job.log
 (1/1) perf_duplicate_probe.py:PerfProbe.test_probe: FAIL: perf probe is placing multiple probe at the same location  (2.28 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2020-07-21T01.29-4f358eb/results.html
JOB TIME   : 2.39 s
[root@ltczz256-lp3 perf]#

**Comment by shirishaganta** [comment]
After Patch:
[root@ltczz256-lp3 perf]# avocado run perf_duplicate_probe.py
/usr/local/lib/python3.6/site-packages/avocado_framework-80.0-py3.6.egg/avocado/plugins/run.py:285: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp. 
  FutureWarning)
JOB ID     : 6da459aadc798a1768fb04763d3095713e71c5ef
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2020-07-21T02.20-6da459a/job.log
 (1/1) perf_duplicate_probe.py:PerfProbe.test_probe: FAIL: perf probe is placing multiple probe at the same location  (2.31 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2020-07-21T02.20-6da459a/results.html
JOB TIME   : 2.43 s
[root@ltczz256-lp3 perf]#

**Comment by shirishaganta** [comment]
In recent Rhel release, many perf related changes went in and the probe point 15 was worked earlier and now because of this changes we changed the probe point to 0 in order to work successfully

**Comment by PraveenPenguin** [comment]
> In recent Rhel release, many perf related changes went in and the probe point 15 was worked earlier and now because of this changes we changed the probe point to 0 in order to work successfully

@shirishaganta @harish-24 as this change from kernel so why not we control probe from YAML that time we need to change python code as changing python code frequent not a good practice and I have the same thought for other PR as well

**Comment by harish-24** [comment]
@shirishaganta any updates here?

**Comment by shirishaganta** [comment]
> > In recent Rhel release, many perf related changes went in and the probe point 15 was worked earlier and now because of this changes we changed the probe point to 0 in order to work successfully
> 
> @shirishaganta @harish-24 as this change from kernel so why not we control probe from YAML that time we need not change python code as changing python code frequent not a good practice and I have the same thought for other PR as well

@PraveenPenguin @harish-24 perf event is 0 for both rhel and suse. And also this probe point will not change frequently. So I dont think we need cfg file for this

**Comment by PraveenPenguin** [comment]
> > > In recent Rhel release, many perf related changes went in and the probe point 15 was worked earlier and now because of this changes we changed the probe point to 0 in order to work successfully
> > 
> > 
> > @shirishaganta @harish-24 as this change from kernel so why not we control probe from YAML that time we need not change python code as changing python code frequent not a good practice and I have the same thought for other PR as well
> 
> @PraveenPenguin @harish-24 perf event is 0 for both rhel and suse. And also this probe point will not change frequently. So I dont think we need cfg file for this

@shirishaganta sure i fine with it

**Review by PraveenPenguin — APPROVED** [review]
LGTM

File: perf/perf_duplicate_probe.py [hunk 1]
```diff
@@ -45,14 +45,16 @@ def setUp(self):
...
+        output1 = output.stderr.decode("utf-8")
+        fail_flag = 0
+        if 'select_task_rq_fair' in output1 and 'select_task_rq_fair_' in output1:
+            fail_flag += 1
         output = genio.read_all_lines("/sys/kernel/debug/tracing/kprobe_events")
         for line in output:
             if 'select_task_rq_fair' in line or 'select_task_rq_fair_' in line:
                 fail_flag += 1
-        if(fail_flag >= 2):
+        if(fail_flag >= 1):
```
Comment (harish-24) [inline perf/perf_duplicate_probe.py:57]: this can be `if fail_flag:`

File: perf/perf_duplicate_probe.py [hunk 2]
```diff
@@ -45,14 +45,16 @@ def setUp(self):
                 self.cancel('%s is needed for the test to be run' % package)
 
     def test_probe(self):
-        output = process.run("perf probe select_task_rq_fair:15", sudo=True)
-        if 'select_task_rq_fair' in output.stderr.decode("utf-8") and 'select_task_rq_fair_' in output.stderr.decode("utf-8"):
-            fail_flag = 1
+        output = process.run("perf probe select_task_rq_fair:0", sudo=True)
+        output1 = output.stderr.decode("utf-8")
```
Comment (harish-24) [inline perf/perf_duplicate_probe.py:49]: can you name this variable better or just replace the output if its not used elsewhere?

File: perf/perf_duplicate_probe.py [hunk 3]
```diff
@@ -45,14 +45,16 @@ def setUp(self):
...
+        fail_flag = 0
```
Comment (harish-24) [inline perf/perf_duplicate_probe.py:50]: this can be a boolean if we are using it as a flag

**Review by harish-24 — CHANGES_REQUESTED** [review]

File: perf/perf_duplicate_probe.py (hunk 1 above)
Comment (shirishaganta) [reply perf/perf_duplicate_probe.py]: ok will change

File: perf/perf_duplicate_probe.py (hunk 3 above)
Comment (shirishaganta) [reply perf/perf_duplicate_probe.py]: ok

File: perf/perf_duplicate_probe.py (hunk 2 above)
Comment (shirishaganta) [reply perf/perf_duplicate_probe.py]: ok

**Comment by shirishaganta** [comment]
[root@ltcfleet8-lp2 perf]# avocado run perf_duplicate_probe.py
JOB ID     : ab4df1bc94e0e422e95491502ff5097284191d2b
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2020-10-15T02.58-ab4df1b/job.log
 (1/1) perf_duplicate_probe.py:PerfProbe.test_probe: FAIL: perf probe is placing multiple probe at the same location  (1.61 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2020-10-15T02.58-ab4df1b/results.html
JOB TIME   : 2.45 s
[root@ltcfleet8-lp2 perf]#

**Review by harish-24 — APPROVED** [review]
LGTM

**Pull request merged**